up-l1/mqtt_5.adoc

```diff
@@ -0,0 +1,276 @@
+    name: "Vehicle_1",
+    domain: veh.com,
+  },
+  entity: UEntity {
+    version_major: 1,
+    id: 1234,
+  },
+  resource: UResource {
+    id: 5678,
+  }
+}
+
+sink:
+UUri {
+  authority: UAuthority {
+    name: "Vehicle_2",
+    domain: veh.com,
+  },
+  entity: UEntity {
+    version_major: 1,
+    id: 4321,
+  },
+  resource: UResource {
+    id: 8765,
+  }
+}
+----
+topic messages are published to: `veh.com/Vehicle_1/veh.com:Vehicle_2/4321/1/8765`
+
+to register a listener with this topic, the following source and sink filter would need to be provided:
+
+[source]
+----
+source filter:
+UUri {
+  authority: UAuthority {
+    name: "Vehicle_1",
+    domain: veh.com,
+  },
+  entity: empty,
+  resource: empty,
+}
+
+sink filter:
+UUri {
+  authority: UAuthority {
+    name: "Vehicle_2",
+    domain: veh.com,
+  },
+  entity: UEntity {
+    version_major: 1,
+    id: 4321,
+  },
+  resource: UResource {
+    id: 8765,
+  }
+}
+----
+
+==== Example 2: Publish Message Type
+
+[source]
+----
+source:
+UUri {
+  authority: UAuthority {
+    name: "Vehicle_1",
+    domain: veh.com,
+  },
+  entity: UEntity {
+    version_major: 1,
+    id: 1234,
+  },
+  resource: UResource {
+    id: 5678,
+  }
+}
+
+sink: None
+----
+
+topic messages are published to: `veh.com/Vehicle_1//1234/1/5678`
+
+to register a listener with this topic, the following source filter would need to be provided:
+
+[source]
+----
+source filter:
+UUri {
+  authority: UAuthority {
+    name: "Vehicle_1",
+    domain: veh.com,
+  },
+  entity: UEntity {
+    version_major: 1,
+    id: 1234,
+  },
+  resource: UResource {
+    id: 5678,
+  }
+}
+----
+
+==== Example 3: Subscribe to all incoming messages to a UAuthority
+
+[source]
+----
+source filter: empty
+
+sink filter:
+UUri {
+  authority: UAuthority {
+    name: "Vehicle_1",
+    domain: veh.com,
+  },
+  entity: empty,
+  resource: empty,
+}
+----
+
+topic: `+/+/veh.com:Vehicle_1/+/+/+`
+
+==== Example 4: Subscribe to all publish messages from a UAuthority
+
+[source]
+----
+source filter:
+UUri {
+  authority: UAuthority {
+    name: "Vehicle_1",
+    domain: veh.com,
+  },
+  entity: empty,
+  resource: empty,
+}
+
+sink filter: None
+----
+
+topic: `veh.com/Vehicle_1//+/+/+`
+
+==== Example 5: Subscribe to all publish messages from a domain
+
+[source]
+----
+source filter:
+UUri {
+  authority: UAuthority {
+    name: empty,
+    domain: veh.com,
+  },
+  entity: empty,
+  resource: empty,
+}
+
+sink filter: None
+----
+
+topic: `veh.com/+//+/+/+`
+
+==== Example 6: Subscribe to all messages sent from a UAuthority
+
+This would include all message types.
+
+[source]
+----
+source filter:
+UUri {
+  authority: UAuthority {
+    name: "Vehicle_1",
+    domain: veh.com,
+  },
+  entity: empty,
+  resource: empty,
+}
+
+sink filter: empty
+----
+
+topic: `veh.com/Vehicle_1/+/+/+/+`
+
+=== Payload Encoding
+
+The MQTT payload **MUST** be a UPayload that is represented as a byte array to reduce size.
```

Comment:
Does the payload need to be encoded in a specific format (base64 encoded) or can it be binary?

Reply:
That is an excellent question. I believe that base64 encoding is used in the rust implementation. But I am not 100% sure: https://docs.rs/protobuf/latest/protobuf/trait.Message.html#method.write_to_bytes. @PLeVasseur do you happen to know if this is for sure base 64 encoding?

Reply:
The payload of an MQTT message is just an array of bytes. No need to base64 encode it.

Reply:
We will pass the upayload.format in the userProperty header ("upayloadformat", ENUM). upayload.data will be passed in the message as bytes.

Reply:
Added to header definition